pact/Pact/Core/IR/Eval/CEK.hs

```diff
@@ -1547,28 +1570,30 @@ applyContToValue (ObjC env info currfield fs vs cont) handler v = do
       returnCEKValue cont handler (VObject (M.fromList (reverse fields)))
 
 applyContToValue (EnforceErrorC info _) handler v = case v of
-  VString err -> returnCEK Mt handler (VError err info)
-  -- TODO: remove
-  _ -> failInvariant info $ "enforce function did not return a string" <> T.pack (show v)
+  VString err ->
+    returnCEKError info Mt handler $ UserEnforceError err
+  VPactValue v' -> throwExecutionError info $ ExpectedStringValue v'
+  _ -> throwExecutionError info $ ExpectedPactValue
 -- Discard the value of running a user guard, no error occured, so
 applyContToValue (IgnoreValueC v cont) handler _v =
   returnCEKValue cont handler (VPactValue v)
 
 applyContToValue (StackPopC i mty cont) handler v = do
-  v' <- (\pv -> maybeTCType i pv mty) =<< enforcePactValue i v
+  v' <- enforcePactValue i v
+  maybeTCType i mty v'
```

Comment:
Should this be moved out to a separate function to avoid having to call `maybeTCType` (which can be forgotten)?

Reply:
I made an asana ticket for this